diff --git a/sites/all/modules/contrib/user_registrationpassword/user_registrationpassword.pages.inc b/sites/all/modules/contrib/user_registrationpassword/user_registrationpassword.pages.inc
index bfc12c8..378247f 100644
--- a/sites/all/modules/contrib/user_registrationpassword/user_registrationpassword.pages.inc
+++ b/sites/all/modules/contrib/user_registrationpassword/user_registrationpassword.pages.inc
@@ -57,15 +57,32 @@ function user_registrationpassword_confirm_account($form, &$form_state, $uid, $t
         watchdog('user', 'User %name used one-time login link at time %timestamp.', array('%name' => $account->name, '%timestamp' => $date));
         // Activate the user and update the access and login time to $current.
         $account = user_save($account, array(
-          'status' => 1,
-          'access' => $current,
-          'login' => $current)
+            'status' => 1,
+            'access' => $current,
+            'login' => $current)
         );
         // Set the new user.
         $user = $account;
+
+        //update user status on master site.
+        module_load_include('inc', 'fs_user', 'fs_user');
+        $result = fs_user_status_update($account->name, 1);
+
+        if (isset($result['success']) && !$result['success']) {
+          drupal_set_message($result['message'], 'error');
+          user_registrationpassword_set_message('linkerror', TRUE);
+        }
+
+        //login in user right now;
+        fs_user_external_login_register($account->name, $account->mail, 'fs_user');
+
+        module_load_include('inc', 'fs_umail', 'fs_umail');
+        $umail = new UMail();
+        $umail->fs_umail_user_confirmation_email($account);
+
         // user_login_finalize() also updates the login timestamp of the
         // user, which invalidates further use of the one-time login link.
-        user_login_finalize();
+        //user_login_finalize();
 
         // Trigger a rules event.
         // @see http://drupal.org/node/1776286
@@ -104,4 +121,4 @@ function user_registrationpassword_confirm_account($form, &$form_state, $uid, $t
       user_registrationpassword_set_message('linkerror', TRUE);
     }
   }
-}
+}